go/libraries/doltcore/env/environment.go

```diff
@@ -175,11 +175,8 @@ func loadWithFormat(ctx context.Context, hdp HomeDirProvider, fs filesys.Filesys
 	if rsErr == nil && dbLoadErr == nil && dbFormatErr == nil {
 		// If the working set isn't present in the DB, create it from the repo state. This step can be removed post 1.0.
 		_, err := dEnv.WorkingSet(ctx)
-		if err == doltdb.ErrWorkingSetNotFound {
-			err := dEnv.initWorkingSetFromRepoState(ctx)
-			if err != nil {
```

Comment:
This was the place were we were getting the error when we wanted to use any dolt command after deleting current branch on CLI. This code tries to init the working set from RepoState if the current working set is invalid/not found. This initially was for moving or changing the initial the `.dolt/` directory into different `./dolt` directory that would have different working set. In that case this code will init the working set correctly to new one, otherwise, the working set is not found which is the case of current head is deleted (I set the RepoState head to empty branch name, which is stored as `ref/heads/` ). This means the working set will be set if it finds any, if not it will be empty.